client/CVideoHandler.cpp

```diff
@@ -266,24 +267,28 @@ bool CVideoPlayer::nextFrame()
 			if (packet.stream_index == stream)
 			{
 				// Decode video frame
-
-				avcodec_decode_video2(codecContext, frame, &frameFinished, &packet);
-
+				int rc = avcodec_send_packet(codecContext, &packet);
+				if (rc >=0)
+					packet.size = 0;
+				rc = avcodec_receive_frame(codecContext, frame);
+				if (rc >= 0)
+					frameFinished = 1;
 				// Did we get a video frame?
 				if (frameFinished)
 				{
-					AVPicture pict;
+					uint8_t *data[4];
+					int linesize[4];
 
 					if (texture) {
-						avpicture_alloc(&pict, AV_PIX_FMT_YUV420P, pos.w, pos.h);
+						av_image_alloc(data, linesize, pos.w, pos.h, AV_PIX_FMT_YUV420P, 1);
 
 						sws_scale(sws, frame->data, frame->linesize,
-								  0, codecContext->height, pict.data, pict.linesize);
+								  0, codecContext->height, data, linesize);
 
-						SDL_UpdateYUVTexture(texture, NULL, pict.data[0], pict.linesize[0],
-								pict.data[1], pict.linesize[1],
-								pict.data[2], pict.linesize[2]);
-						avpicture_free(&pict);
+						SDL_UpdateYUVTexture(texture, NULL, data[0], linesize[0],
+								data[1], linesize[1],
+								data[2], linesize[2]);
+						av_freep(&data[0]);

```

Comment:
Why do we free only data[0] if we also allocated data[1] and data[2] and they are pointers as well?

Reply:
I see it is according to docs